### PR TITLE
[Darwin] Duplicate / begin raising `MTRDeviceController`

### PR DESCRIPTION
**Pull request by kiel-apple**
similar to what we just did with `MTRDevice`.  prepare to add `XPC` version.

### CODE REVIEW

**Comment by kiel-apple** [comment]
the interesting diff so far is: https://github.com/project-chip/connectedhomeip/pull/34832/commits/0df0bad8bd79acfb6c4e12068a5503520bd71fae

**Comment by kiel-apple** [comment]
closing this for a hot second, going to get the duplication completed and avoid wasting runners during their time of need

File: src/darwin/Framework/CHIP/MTRDeviceControllerFactory.h [hunk 1]
```diff
@@ -31,7 +32,7 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol MTROTAProviderDelegate;
 @protocol MTRKeypair;
 
-@class MTRDeviceController;
+//@class MTRDeviceController;
```
Comment (woody-apple) [inline src/darwin/Framework/CHIP/MTRDeviceControllerFactory.h:35]:
```suggestion
```

File: src/darwin/Framework/CHIP/MTRDeviceController.h [hunk 2]
```diff
@@ -57,7 +57,7 @@ MTR_AVAILABLE(ios(16.1), macos(13.0), watchos(9.1), tvos(16.1))
  * Once this returns non-nil, it's the caller's responsibility to call shutdown
  * on the controller to avoid leaking it.
  */
-- (nullable instancetype)initWithParameters:(MTRDeviceControllerAbstractParameters *)parameters
+- (nullable MTRDeviceController *)initWithParameters:(MTRDeviceControllerAbstractParameters *)parameters
```
Comment (woody-apple) [inline src/darwin/Framework/CHIP/MTRDeviceController.h:60]: Is this still needed?

**Review by woody-apple — APPROVED** [review]

**Pull request merged**